yewprint-doc/src/tag/example.rs

```diff
@@ -0,0 +1,84 @@
+use yew::prelude::*;
+use yewprint::{Tag, IconName,Intent, ConditionalClass};
+
+pub struct Example {
+    props: ExampleProps,
+    link: ComponentLink<Self>,
+}
+
+#[derive(Clone, PartialEq, Properties)]
+pub struct ExampleProps {
+    pub initial_tags: Vec<String>,
+    pub tags: Vec<String>,
+    pub active: bool,
+    pub fill: bool,
+    pub icon: ConditionalClass,
+    pub intent: Option<Intent>,
+    pub interactive: bool,
+    pub large: bool,
+    pub minimal: bool,
+    pub multiline: bool,
+    //onClick,
+    pub removable: ConditionalClass,
+    pub right_icon: ConditionalClass,
+    pub round: bool,
+}
+
+pub enum ExampleMsg {
+    Remove(String),
+}
+
+
+impl Component for Example {
+    type Message = ExampleMsg;
+    type Properties = ExampleProps;
+
+    fn create(props: Self::Properties, link: ComponentLink<Self>) -> Self {
+        Example { props, link }
+    }
+
+    fn update(&mut self, msg: Self::Message) -> ShouldRender {
+        let ExampleMsg::Remove(label) = msg;
+        self.props.tags = self.props.tags.clone().into_iter().filter(|l| *l != label).collect();
+        true
+    }
+
+    fn change(&mut self, props: Self::Properties) -> ShouldRender {
+        if self.props != props {
+            self.props = props;
+            true
+        } else {
+            false
+        }
+    }
+
+    fn view(&self) -> Html {
+        let tags = self.props.tags.clone().into_iter().map(|label| {
+            let label2 = label.clone();
+            let remove = self.props.removable.map_some(self.link.callback(move |_| ExampleMsg::Remove(label2.clone()))); 

```

Comment:
In general in Rust when you want to avoid "variable2" in order to move it in a closure you can use this pattern instead:

```suggestion
            let remove = {
                let label = label.clone();
                self.props.removable.map_some(self.link.callback(move |_| ExampleMsg::Remove(label)))
            };
```

Reply:
Looks a bit better indeed. Still I don't get how having this supplementary storage for a `label.clone()` is significantly different from the same thing without this storage:

With `.iter()` I get the following error:

```
error[E0495]: cannot infer an appropriate lifetime for lifetime parameter in function call due to conflicting requirements
  --> yewprint-doc/src/tag/example.rs:56:36
   |
56 |         let tags = self.props.tags.iter().map(|label| {
   |                                    ^^^^
   |
note: first, the lifetime cannot outlive the anonymous lifetime #1 defined on the method body at 55:5...
  --> yewprint-doc/src/tag/example.rs:55:5
   |
55 | /     fn view(&self) -> Html {
56 | |         let tags = self.props.tags.iter().map(|label| {
57 | |             let remove = {
58 | |                // let label = label.clone();
...  |
84 | |         }
85 | |     }
   | |_____^
note: ...so that reference does not outlive borrowed content
  --> yewprint-doc/src/tag/example.rs:56:20
   |
56 |         let tags = self.props.tags.iter().map(|label| {
   |                    ^^^^^^^^^^^^^^^
   = note: but, the lifetime must be valid for the static lifetime...
note: ...so that the type `[closure@yewprint-doc/src/tag/example.rs:59:66: 59:108 label:&std::string::String]` will meet its required lifetime bounds
  --> yewprint-doc/src/tag/example.rs:59:57
   |
59 |                 self.props.removable.map_some(self.link.callback(move |_| ExampleMsg::Remove(label.clone())))
   |                                                         ^^^^^^^^
```

I get different errors depending on `iter()` `.clone().into_iter()` when this `let label = label.clone()` is not outside the closure.